### PR TITLE
Fix upgrade script to print errors if checks fail after upgrade

### PR DESCRIPTION
**Pull request by amitaekbote**
## High-level description

Fixes a bug in the upgrade script which does not print errors if checks fail after an upgrade

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

https://jira.mesosphere.com/browse/DCOS_OSS-2028

## Checklist for all PRs

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-teamcity** [comment]
Corresponding DC/OS PR: https://github.com/dcos/dcos/pull/2465

**Comment by amitaekbote** [comment]
@branden @mnaboka this is ready for review. 

Tested out on a node

```core@ip-10-0-6-9 ~ $ bash test.sh 

Node upgrade not successful, checks failed
{
  "status": 1,
  "checks": {
    "clock_sync": {
      "output": "Clock is synced\n",
      "status": 0
    },
    "components_master": {
      "output": "",
      "status": 0
    },
    "curl": {
      "output": "",
      "status": 0
    },
    "ip_detect_script": {
      "output": "10.0.6.9 is a valid IPV4 address\n",
      "status": 0
    },
    "journald_dir_permissions": {
      "output": "time=\"2018-03-09T19:27:51Z\" level=fatal msg=\"Error executing systemd journal check: directory /var/log/journal has wrong permissions: group r-- bit must be set. \n\nTry to run: systemd-tmpfiles --create --prefix /var/log/journal\" \n",
      "status": 1
    },
    "mesos_master_replog_synchronized": {
      "output": "",
      "status": 0
    },
    "tar": {
      "output": "",
      "status": 0
    },
    "unzip": {
      "output": "",
      "status": 0
    },
    "xz": {
      "output": "",
      "status": 0
    }
  }
}```

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

File: dcos_installer/upgrade.py [hunk 1]
```diff
@@ -130,6 +130,7 @@
 
 if [[ "$SKIP_CHECKS" = "false" ]]; then
     T=300
+    set +e
```
Comment (branden) [inline dcos_installer/upgrade.py]: Let's use the long option form: `set +o errexit`. It's more readable, and it's consistent with the other uses of `set` in this script.

Comment (amitaekbote) [reply dcos_installer/upgrade.py]: done

**Comment by amitaekbote** [comment]
@mesosphere-mergebot bump-ee

**Comment by amitaekbote** [comment]
@mesosphere-mergebot ship_it_label_status_check

**Pull request merged**